test/contrasts.jl

```diff
@@ -0,0 +1,92 @@
+module TestContrasts
+
+using Base.Test
+using DataFrames
+
+
+d = DataFrame(x = @pdata( [:a, :b, :c, :a, :a, :b] ))
+
+mf = ModelFrame(Formula(Nothing(), :x), d)
+
+@test ModelMatrix(mf).m == [1  0  0
+                            1  1  0
+                            1  0  1
+                            1  0  0
+                            1  0  0
+                            1  1  0]
+@test coefnames(mf) == ["(Intercept)"; "x: b"; "x: c"]
+
+contrast!(mf, x = SumContrasts)
+@test ModelMatrix(mf).m == [1 -1 -1
+                            1  1  0
+                            1  0  1
+                            1 -1 -1
+                            1 -1 -1
+                            1  1  0]
+@test coefnames(mf) == ["(Intercept)"; "x: b"; "x: c"]
```

Comment:
That is in fact what R does. But I've always found it a little odd that `contr.treatment` and `contr.helmert` go the other way. So while it wouldn't be technically that difficult to change, I'd prefer to keep the default for the base level as consistent as possible across types.